core/DataTable/Filter/ColumnDelete.php

```diff
@@ -181,8 +181,7 @@ protected function removeColumnsFromTable(&$table)
                 unset($table[$index][$column]);
             }
 
-            // Restore me in Piwik 4
-            //$this->removeColumnsFromTable($row);
+            $this->removeColumnsFromTable($row);
```

Comment:
That actually doesn't work for arrays, as `$row` is no reference, but a 'copy', so the filter isn't applied correctly for nested arrays. I'll apply a fix